atef/procedure.py

```diff
@@ -21,40 +21,41 @@
 import pathlib
 from dataclasses import dataclass, field
 from typing import (Any, Dict, Generator, List, Literal, Optional, Sequence,
-                    Union, cast)
+                    Tuple, Union, cast)
 
 import apischema
-import databroker
 import ophyd
+import pandas as pd
 import yaml
-from bluesky import RunEngine
+from bluesky_queueserver.manager.profile_ops import (
+    existing_plans_and_devices_from_nspace, validate_plan)
 
 from atef import util
-from atef.cache import _SignalCache, get_signal_cache
+from atef.cache import DataCache, _SignalCache, get_signal_cache
 from atef.check import Comparison
 from atef.config import (ConfigurationFile, PreparedComparison, PreparedFile,
                          PreparedSignalComparison, run_passive_step)
-from atef.enums import GroupResultMode, Severity
+from atef.enums import GroupResultMode, PlanDestination, Severity
 from atef.exceptions import PreparedComparisonException
+from atef.plan_utils import (BlueskyState, GlobalRunEngine,
+                             get_default_namespace, register_run_identifier,
+                             run_in_local_RE)
+from atef.reduce import ReduceMethod
 from atef.result import Result, _summarize_result_severity, incomplete_result
-from atef.type_hints import AnyPath, Number, PrimitiveType
+from atef.type_hints import AnyDataclass, AnyPath, Number, PrimitiveType
 from atef.yaml_support import init_yaml_support
 
 from . import serialization
 
 logger = logging.getLogger(__name__)
 
 
-class BlueskyState:
-    def __new__(cls):
-        if not hasattr(cls, 'instance'):
-            cls.instance = super(BlueskyState, cls).__new__(cls)
-        return cls.instance
+# BlueskyState tracker.  Keys are the id of the top-level ProcedureFile
+# Includes one default BlueskyState, with key id(None)
+BS_STATE_MAP: Dict[int, BlueskyState] = {}
 
-    def __init__(self):
-        self.db = databroker.Broker.named('temp')
-        self.RE = RunEngine({})
-        self.RE.subscribe(self.db.insert)
+# Max depth for plan steps
```

Comment:
Depth of 2 would be "plan run inside a plan"?

Reply:
I was envisioning plan steps inside multiple `ProcedureGroup`s.